changes capture_test db to capture_dev

KasiaMisirli opened the pull request:
Co-authored-by: Jake Pitman <jakeseanp@gmail.com>
Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>

KasiaMisirli committed:
changes capture_test db to capture_dev

Co-authored-by: Jake Pitman <jakeseanp@gmail.com>
Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>

radar commented on config/test.exs:
```diff
@@ -1,6 +1,6 @@
 
 config :capture, Capture.Repo,
-  database: "capture_test",
+  database: "capture_dev",
```
This should be left as `capture_test`, because this file is the configuration file for the test environment.

cset00 replied:
Hey Ryan, is there an ecto way to create the `capture_test` database and insert the table into it?

We created a `capture_test` database, and created the responses table, but the table keeps going to `capture_dev`

radar commented on lib/capture/response.ex:
```diff
@@ -2,12 +2,12 @@ defmodule Capture.Response do
-    field :neutral, :integer
-    field :agree, :integer
-    field :strongly_agree, :integer
+    field(:survey_id, :integer)
```
I am a little confused about the brackets being added here. Was that a suggestion made by `mix format`?

cset00 replied:
Probably yup.

radar commented on test/capture_web/router_test.exs:
```diff
@@ -40,23 +42,29 @@ defmodule CaptureWeb.RouterTest do
+      response =
+        Response
+        |> Responses.for_survey(1)
+        |> IO.inspect(label: "before")
```
Good use of the `label` option for `IO.inspect`!

radar commented on test/capture_web/router_test.exs:
```diff
@@ -23,15 +23,17 @@ defmodule CaptureWeb.RouterTest do
-      |> Responses.for_survey(1)
-      |> Responses.for_question(1)
-      |> Repo.one
+      response =
```
One way to tidy up these tests might be to create a function for finding a particular response row. I would suggest giving this a shot and see if you can do it!

cset00 replied:
Sounds good - thanks!

cset00 committed:
Create test for counter

Co-authored-by: Kasia Misirli <kasia.misirli@gmail.com>
Co-authored-by: Jake Pitman <jakeseanp@gmail.com>

JakePitman committed:
Handle updated responses by incrementing & decrementing

Co-authored-by: Kasia Misirli <kasia.misirli@gmail.com>
Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>

KasiaMisirli committed:
add capture_test db

Co-authored-by: Jake Pitman <jakeseanp@gmail.com>
Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>

KasiaMisirli committed:
WIP 3/4 test passing

Co-authored-by: Jake Pitman <jakeseanp@gmail.com>
Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>

JakePitman committed:
update to new db structure

The old interpretation of response_id was that it
stored one of the 5 possible response values that
can be selected by a user. This new structure
interprets response_id as a unique identifier,
and value as the user's selection

Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>
Co-authored-by: Kasia Misirli <kasia.misirli@gmail.com>

KasiaMisirli committed:
add plug and jason to parse json body

Co-authored-by: Jake Pitman <jakeseanp@gmail.com>
Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>

JakePitman committed:
cool beans it works

changed survey_id, question_id & response_id to
strings from integers in database structure

Co-authored-by: Kasia Misirli <kasia.misirli@gmail.com>
Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>

cset00 committed:
Testing get API to return query results.

Co-authored-by: Kasia Misirli <kasia.misirli@gmail.com>
Co-authored-by: Jake Pitman <jakeseanp@gmail.com>

cset00 committed:
Yayy count works! WIP anyway.

Co-authored-by: Kasia Misirli <kasia.misirli@gmail.com>
Co-authored-by: Jake Pitman <jakeseanp@gmail.com>

KasiaMisirli committed:
Getting all responses per survey

Co-authored-by: Jake Pitman <jakeseanp@gmail.com>
Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>

KasiaMisirli committed:
Add query for each question in a survey

Co-authored-by: Jake Pitman <jakeseanp@gmail.com>
Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>

JakePitman committed:
run mix format

Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>
Co-authored-by: Kasia Misirli <kasia.misirli@gmail.com>

JakePitman committed:
add query_selector function to check if question_id is in params

Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>
Co-authored-by: Kasia Misirli <kasia.misirli@gmail.com>

JakePitman committed:
refactor query_selector into two functions

this way we can pattern match that a question_id
has been given as a parameter

Co-authored-by: Caroline Setiawan <vc.setiawan@gmail.com>
Co-authored-by: Kasia Misirli <kasia.misirli@gmail.com>

JakePitman committed:
send response_body as json in get /responses

radar commented on test/capture_web/router_test.exs:
```diff
@@ -8,55 +8,64 @@ defmodule CaptureWeb.RouterTest do
+        |> Responses.for_survey(1)
+        |> Responses.for_question(1)
+        |> Repo.one()
+        |> IO.inspect(label: "blah")
```
You might want to remove this `IO.inspect` if you don't need it
